python/paddle/v2/fluid/tests/test_ftrl_op.py

```diff
@@ -0,0 +1,58 @@
+import unittest
+import numpy as np
+from op_test import OpTest
+
+
+class TestFTRLOp(OpTest):
+    def setUp(self):
+        self.op_type = "ftrl"
+        w = np.random.random((102, 105)).astype("float32")
+        g = np.random.random((102, 105)).astype("float32")
+        sq_accum = np.full((102, 105), 0.1).astype("float32")
+        linear_accum = np.full((102, 105), 0.1).astype("float32")
+        lr = np.array([0.1]).astype("float32")
+        l1 = 0.1
+        l2 = 0.2
+        lr_power = -0.5
+
+        self.inputs = {
+            'Param': w,
+            'SquaredAccumulator': sq_accum,
+            'LinearAccumulator': linear_accum,
+            'Grad': g,
+            'LearningRate': lr
+        }
+        self.attrs = {'l1': l1, 'l2': l2, 'lr_power': lr_power}
+        new_accum = sq_accum + g * g
+        if lr_power == -0.5:
+            linear_out = linear_accum + g
+            -((np.sqrt(new_accum) - np.sqrt(sq_accum)) / lr) * w
+        else:
+            linear_out = linear_accum + g
+            -((np.power(new_accum, -lr_power) - np.power(sq_accum, -lr_power)) /
+              lr) * w
+
+        x = (l1 * np.sign(linear_out) - linear_out)
+        if lr_power == -0.5:
+            y = (np.sqrt(new_accum) / lr) + (2 * l2)
+            pre_shrink = x / y
+            param_out = np.where(linear_out > l1, pre_shrink, 0)
```

Comment:
I compared the result with the tensorflow, this part is wrong.

Reply:
Oh I see, yes the C++ code is correct and is same as Tensorflow. Let me check the python code again.

Reply:
I commented out the part for calculating `param_out` and even then, the computation of `linear_out` itself doesn't match the C++ output. I think it could be a precision error, I can try once with double.